contracts/protocol/pool/PoolInstantWithdraw.sol

```diff
@@ -0,0 +1,571 @@
+                msg.sender,
+                borrowAmount
+            );
+
+            // update borrow asset interest rate
+            reserve.updateInterestRates(reserveCache, borrowAsset, 0, 0);
+        }
+
+        // create Loan
+        uint256 loanId = ps._loanIdCounter.current();
+        ps._loanIdCounter.increment();
+        ps._termLoans[loanId] = DataTypes.TermLoanData({
+            loanId: loanId,
+            state: DataTypes.LoanState.Active,
+            startTime: uint40(block.timestamp),
+            endTime: uint40(
+                IInstantNFTOracle(WITHDRAW_ORACLE).getEndTime(collateralTokenId)
+            ),
+            borrower: msg.sender,
+            collateralAsset: collateralAsset,
+            collateralTokenId: collateralTokenId.toUint64(),
+            borrowAsset: borrowAsset,
+            borrowAmount: borrowAmount,
+            stableRate: nextBorrowRate,
+            discountRate: discountRate
+        });
+        ps._userLoanIds[msg.sender].push(loanId);
+
+        emit Borrow(
+            borrowAsset,
+            VAULT_CONTRACT,
+            VAULT_CONTRACT,
+            borrowAmount,
+            DataTypes.InterestRateMode.STABLE,
+            nextBorrowRate,
+            referralCode
+        );
+        emit LoanCreated(
+            msg.sender,
+            loanId,
+            collateralAsset,
+            collateralTokenId,
+            borrowAsset,
+            borrowAmount,
+            nextBorrowRate,
+            discountRate
+        );
+    }
+
+    /// @inheritdoc IPoolInstantWithdraw
+    function swapLoanCollateral(uint256 loanId, address receiver)
+        external
+        override
+        nonReentrant
+    {
+        DataTypes.PoolStorage storage ps = poolStorage();
+        DataTypes.TermLoanData storage loan = ps._termLoans[loanId];
+        // check loan state
+        require(
+            loan.state == DataTypes.LoanState.Active,
+            Errors.INVALID_LOAN_STATE
+        );
+
+        address collateralAsset = loan.collateralAsset;
+        uint256 collateralTokenId = uint256(loan.collateralTokenId);
+        address borrowAsset = loan.borrowAsset;
+        //here oracle need to guarantee presentValue > debtValue.
+        uint256 presentValue = IInstantNFTOracle(WITHDRAW_ORACLE)
+            .getPresentValueByDiscountRate(
+                collateralTokenId,
+                loan.discountRate
+            );
+        uint256 borrowAssetAmountNeeded = _calculateAssetAmountByETHValue(
+            borrowAsset,
+            presentValue
+        );
+
+        // update borrow asset state
+        DataTypes.ReserveCache memory reserveCache;
+        DataTypes.ReserveData storage reserve = ps._reserves[borrowAsset];
+        reserveCache = reserve.cache();
+        reserve.updateState(reserveCache);
+
+        // repay borrow asset debt and update interest rate
+        uint256 totalDebt = _calculateTotalStableDebt(
```

Comment:
how about `_calculateLoanStableDebt`
and the var `loanDebt` or `loanStableDebt`